apps/nextjs/src/components/ContextProviders/ChatProvider.tsx

```diff
@@ -107,34 +60,28 @@ export function ChatProvider({ id, children }: Readonly<ChatProviderProps>) {
   const { invokeActionMessages } = useActionMessages();
 
   /******************* Streaming of all chat starts from messages here *******************/
-  const initialMessages = useMemo(() => {
-    return chat?.messages?.filter((m) =>
-      isValidMessageRole(m.role),
-    ) as Message[];
-  }, [chat?.messages]);
-
   const {
     messages,
     append,
     reload,
     stop: stopStreaming,
     isLoading,
-    setMessages,
   } = useChat({
     sendExtraMessageFields: true,
+    // TODO: these initial messages are sent and used by the chat endpoint
     initialMessages,
     generateId: () => generateMessageId({ role: "user" }),
     id,
     body: {
       id,
-      lessonPlan: chat?.lessonPlan,
+      // TODO: this lesson plan is sent and used by the chat endpoint
+      lessonPlan,
```

Comment:
Potential race condition here if setting the initial append before lessonPlan has been fetched. But if that's the case then the lessonPlan should be empty anyway

Reply:
It looks like that was already the case with no problems though